main.go

```diff
@@ -255,14 +259,43 @@ func main() {
 	}
 
 	mux := http.FileServer(&S3{client, bucket, bucketPath})
+
+	// Wrap the existing mux with the CORS middleware.
+	opts := cors.Options{
+		AllowOriginFunc: func(origin string) bool {
+			if allowedCorsOrigin == "" {
+				return true
+			}
+			for _, allowedOrigin := range strings.Split(allowedCorsOrigin, ",") {
+				if wildcard.MatchSimple(allowedOrigin, origin) {
+					return true
+				}
+			}
+			return false
+		},
+		AllowedMethods: []string{
+			http.MethodGet,
+			http.MethodPut,
+			http.MethodHead,
+			http.MethodPost,
+			http.MethodDelete,
+			http.MethodOptions,
+			http.MethodPatch,
+		},
+		AllowedHeaders:   []string{"*"},
+		ExposedHeaders:   []string{"*"},
+		AllowCredentials: true,
+	}
+	mux_handler := cors.New(opts).Handler(mux)
```

Comment:
```suggestion
	muxHandler := cors.New(opts).Handler(mux)
```

v_1 is not idiomatic style lets use camel-casing.